bundles/org.openhab.persistence.rrd4j/src/main/java/org/openhab/persistence/rrd4j/internal/RRD4jPersistenceService.java

```diff
@@ -230,73 +230,83 @@ public void store(Item item) {
     @Override
     public Iterable<HistoricItem> query(FilterCriteria filter) {
         String itemName = filter.getItemName();
+
         RrdDb db = getDB(itemName);
-        if (db != null) {
-            ConsolFun consolidationFunction = getConsolidationFunction(db);
-            long start = 0L;
-            long end = filter.getEndDate() == null ? System.currentTimeMillis() / 1000
-                    : filter.getEndDate().toInstant().getEpochSecond();
+        if (db == null) {
+            logger.warn("Could not find item '{}' in rrd4j database", itemName);
```

Comment:
Afaics, we didn't have this warning before - and I'd think that there might be legit situations for a query even if no db file is there (yet).

Reply:
Yes it was meant to be debug level. I'll change that.

Reply:
I only reduced the amount of of nesting/indentation with this quick return statement. That makes the code a bit more readable. Moving logic to other methods also helps with that.

Reply:
Yes, I saw and understood that - although it made it trickier to review and see the parts that are the real change in it.